src/workingtitle-vcockpits-instruments-navsystems/html_ui/WorkingTitle/Pages/VCockpit/Instruments/NavSystems/Shared/WT_MapElement.js

```diff
@@ -0,0 +1,384 @@
+class WT_MapElement extends MapInstrumentElement {
+    // _settings should be an iterable of WT_MapSetting or WT_MapSettingGroup objects
+    constructor(_varNameID, _settings = []) {
+        super();
+        this.varNameID = _varNameID;
+        
+        this._settings = Array.from(_settings);
+        this._settingsToSync = [];
+        for (let setting of this._settings) {
+            if (setting.toSync) {
+                this._settingsToSync.push(setting);
+            }
+        }
+    }
+    
+    addSetting(_setting) {
+        this._settings.push(_setting);
+        if (_setting.toSync) {
+            this._settingsToSync.push(_setting);
+        }
+    }
+    
+    callSettingsOnTemplateLoaded() {
+        for (let setting of this._settings) {
+            setting.onTemplateLoaded();
+        }
+    }
+    
+    onUpdate(_deltaTime) {
+        super.onUpdate(_deltaTime);
+        
+        let initID = WT_MapElement.getSettingVar(WT_MapElement.VARNAME_SYNC_INIT_ROOT_DEFAULT, WT_MapElement.VARNAME_SYNC_ALL_ID, "");
+        if (initID == this.varNameID) {
+            this.syncMasterToAllSettings();
+            WT_MapElement.setSettingVar(WT_MapElement.VARNAME_SYNC_INIT_ROOT_DEFAULT, WT_MapElement.VARNAME_SYNC_ALL_ID, "");
+            WT_MapElement.setSettingVar(WT_MapElement.VARNAME_SYNC_ROOT, WT_MapElement.VARNAME_SYNC_ALL_ID, WT_MapElement.Sync.ALL);
+        }
+        
+        let sync = WT_MapElement.getSettingVar(WT_MapElement.VARNAME_SYNC_ROOT, WT_MapElement.VARNAME_SYNC_ALL_ID);
```

Comment:
I feel like there's some inefficiency here in (possibly) setting the value in the prior if statement then fetching it in this one.   It's small, but it makes me flinch a little.   If `(initID == this.varNameID)` you already know what `WT_MapElement.getSettingVar(WT_MapElement.VARNAME_SYNC_ROOT, WT_MapElement.VARNAME_SYNC_ALL_ID)` is going to be.  Maybe make that a trinary assignment based on whether  `(initID == this.varNameID)`?

Reply:
resolved with 8673401